### DISCUSSION TITLE
Will tokenizer automatically shrink the number of words to fit into the maximum sequence length

### DISCUSSION THREAD

**Discussion by lukemao** [google-research/text-to-text-transfer-transformer#841] [Q&A]
Hello,

If run stsb function of T5 

`
input_ids = tokenizer.encode('stsb sentence1: '+sent1+'. sentence2: '+sent2+'.', return_tensors='tf')
`

If both input sentences, sent1 and sent2 are longer than model_max_length of 512, will the tokenizer shrink the over input length or individual sentence?

Mant thanks,
Lu

**Comment by craffel** [comment]
The tokenizer itself does not do this. This is handled downstream in the input processing pipeline.

**Reply by lukemao** [reply]
thanks @craffel what do you mean by 'handled downstream in the input processing pipeline' ?

**Reply by craffel** [reply]
It happens after tokenization but before the data is fed into the model.

**Reply by lukemao** [reply]
ok I see

**Reply by lukemao** [reply]
will the model trim each input sentenceX to make the whole input fit under the max length 512 ?

**Reply by craffel** [reply]
It will truncate the full tokenized sequence.

**Reply by lukemao** [reply]
ok I see, thanks Colin

**Reply by lukemao** [reply]
I am trying to truncate both input sequence into the same length: 

```
sent1 = ' '.join(input_compl.split()[:100])
sent2 = ' '.join(input_compl.split()[:100])
```

And then run the t5's stsb:

```
input_ids = tokenizer.encode('stsb sentence1: '+sent1+'. sentence2: '+sent2+'.', return_tensors='tf')
outputs = model.generate(input_ids=input_ids, 
                         max_length=300
                        )
float(tokenizer.decode(outputs[0][2]))
```

But, even though I am comparing the same sentences, I am not getting the top score of 5

returns:
```
4.8
```
any idea why?

**Reply by craffel** [reply]
Could be because T5 has never seen any examples where the two sentences are the same.
